rendermodules/point_match_optimization/pt_match_optimization.py

```diff
@@ -0,0 +1,583 @@
+                                                            stack)
+    url_suffix = "render-parameters"
+    if url_options['renderWithFilter'] is False:
+        url_suffix += '?filter=false'
+    else:
+        url_suffix += '?filter=true'
+    if url_options['normalizeForMatching'] is False:
+        url_suffix += '&normalizeForMatching=false'
+    else:
+        url_suffix += '&normalizeForMatching=true'
+    if url_options['renderWithoutMask'] is False:
+        url_suffix += '&renderWithoutMask=false'
+    else:
+        url_suffix += '&renderWithoutMask=true'
+
+    canvas_urls = "%s/%s/%s %s/%s/%s"%(tile_base_url,
+                                        tile1,
+                                        url_suffix,
+                                        tile_base_url,
+                                        tile2,
+                                        url_suffix)
+    return canvas_urls
+'''
+
+def get_parameter_sets_and_strings(SIFT_options):
+    #length = [len(x) for x in SIFT_options]
+    #maxlen = np.max(length)
+
+    new_options = {}
+    all_parameter_list = []
+    for key in SIFT_options:
+        if (~isinstance(SIFT_options[key], list)):
+            new_options[key] = list(SIFT_options[key])
+        else:
+            new_options[key] = SIFT_options[key]
+
+    keys = []
+    for n in new_options:
+        keys.append(n)
+        all_parameter_list.append(new_options[n])
+
+    all_combinations = list(itertools.product(*all_parameter_list))
+    #return all_parameter_list
+    return keys, all_combinations
+
+def render_from_template(directory, template_name, **kwargs):
+    loader = FileSystemLoader(directory)
+    env = Environment(loader=loader)
+    template = env.get_template(template_name)
+    return template.render(**kwargs)
+
+def draw_matches(render, stack, tile1, tile2, ptmatches, scale, outdir, color=None):
+    
+    im1_url = "%s:%d/render-ws/v1/owner/%s/project/%s/stack/%s/tile/%s/png-image?scale=%f&filter=false&normalizeForMatchin=true"%(render.DEFAULT_HOST,
+                    render.DEFAULT_PORT,
+                    render.DEFAULT_OWNER,
+                    render.DEFAULT_PROJECT,
+                    stack,
+                    tile1,
+                    scale)
+    im2_url = "%s:%d/render-ws/v1/owner/%s/project/%s/stack/%s/tile/%s/png-image?scale=%f&filter=false&normalizeForMatchin=true"%(render.DEFAULT_HOST,
+                    render.DEFAULT_PORT,
+                    render.DEFAULT_OWNER,
+                    render.DEFAULT_PROJECT,
+                    stack,
+                    tile2,
+                    scale)
+
+    print(im1_url)
+    print(im2_url)
+    r = urllib.urlopen(im1_url)
+    img1 = np.asarray(bytearray(r.read()), dtype="uint8")
+    img1 = cv2.imdecode(img1, cv2.IMREAD_COLOR)
+
+    r = urllib.urlopen(im2_url)
+    img2 = np.asarray(bytearray(r.read()), dtype="uint8")
+    img2 = cv2.imdecode(img2, cv2.IMREAD_COLOR)
```

Comment:
Don't think it's detrimental, but you should be able to use `renderapi.image.get_tile_image_data` to do something like this as well.